src/indieweb_utils/webmentions/validate.py

```diff
@@ -0,0 +1,157 @@
+from urllib import parse as url_parse
+from typing import List
+
+from bs4 import BeautifulSoup
+import requests
+
+from ..utils.urls import canonicalize_url
+
+
+class WebmentionValidationError(Exception):
+    pass
+
+
+class WebmentionIsGone(Exception):
+    pass
+
+
+def process_vouch(vouch: str, source: str, vouch_list: List[str]) -> bool:
+    # use vouch to flag webmentions for moderation
+    # see Vouch spec for more: https://indieweb.org/Vouch
+    # by default, webmention should require moderation
+    # if a vouch is valid, webmention does not need moderation
```

Comment:
Make this a docstring, rather than comments. That'll make them easier to extract by sphinx, I think.

```suggestion
    """
    use vouch to flag webmentions for moderation
    see Vouch spec for more: https://indieweb.org/Vouch
    by default, webmention should require moderation
    if a vouch is valid, webmention does not need moderation
    """
```